### PR TITLE
Test: batch rpc with params

### PR DESCRIPTION
**Pull request by instagibbs**
Useful as an example and test case.

### CODE REVIEW

File: test/functional/interface_rpc.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def test_batch_request(self):
             # work fine.
             {"method": "invalidmethod", "id": 2},
             # Another call that should succeed.
-            {"method": "getbestblockhash", "id": 3},
+            {"method": "getblockhash", "id": 3, "params": [self.nodes[0].getblockcount()]},
```
Comment (laanwj) [inline test/functional/interface_rpc.py:48]: Concept ACK on testing a call with parameters.
Though, you're calling into the RPC while building the query, which is a bit funny (but not a problem in practice as there is a well-defiend execution order).

Comment (instagibbs) [reply test/functional/interface_rpc.py:48]: doesn't reflect practical usage sure, but was a quick way to demonstrate how it's done, and exercise it :)

I was looking for an example and tests are the first place I look!

Comment (promag) [reply test/functional/interface_rpc.py:48]: Could just use `"params": [0]`.

**Review by promag — COMMENTED** [review]
ACK

File: test/functional/interface_rpc.py (hunk 1 above)
Comment (instagibbs) [reply test/functional/interface_rpc.py]: fair enough that's easy to understand, done

**Review by theStack — APPROVED** [review]
ACK e1fdd2963baab68bb6a77af2ad7a07fcacd4e73e

**Comment by laanwj** [comment]
ACK e1fdd2963baab68bb6a77af2ad7a07fcacd4e73e

**Pull request merged**